src/pages/index.js

```diff
@@ -29,51 +43,124 @@ const renderCard = (item) => {
 };
 
 //create objects
+//api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7ff811c0-9f09-42c2-975e-cca434c12992",
+    "content-type": "application/json",
+    "Access-Control-Allow-Origin": "*",
+  },
+});
+
 //profile section
-const userInformation = new UserInfo(".profile");
+const userInfo = new UserInfo(".profile");
+
+const profileSection = new Section(
+  {
+    renderer: () => {
+      api.getUserInformation().then(({ name, avatar, about, _id }) => {
+        userInfo.setAllInfo({
+          name: name,
+          avatar: avatar,
+          about: about,
+          _id: _id,
+        });
+      });
+    },
+  },
+  ".profile"
+);
 
 //cards section
 const cardList = new Section(
   {
-    data: initCards,
-    renderer: renderCard,
+    renderer: () => {
+      api.getInitialCards().then((items) => {
+        items.forEach((item) => {
+          renderCard(item);
+        });
+      });
+    },
   },
   ".page__cards"
 );
 
 //edit form
-const editFormValidator = new FormValidation("profile-edit-form", formData);
+const editFormValidator = new FormValidation("edit", formData);
 editFormValidator.enableValidation();
 
 const editPopup = new PopupWithForm(".modal_type_edit", (values) => {
-  userInformation.setUserInfo(values);
-  editFormValidator.disableSubmit();
-  editPopup.close();
+  editPopup.toggleSaving();
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```